koku/api/report/queries.py

```diff
@@ -399,80 +437,110 @@ def _set_tag_exclusion_filters(self):
+            # found in the same json structure per row.
+            _exclusion_composed = None
+            for _filt in _filter_list:
+                _filt_composed = QueryFilterCollection([QueryFilter(**_filt)]).compose()
+                if not _exclusion_composed:
+                    _exclusion_composed = _filt_composed
                 else:
-                    tag_exclusion_composed = tag_exclusion_composed & tag_filt_composed
-            if tag_exclusion_composed:
-                tag_exclusion_composed = (
-                    tag_exclusion_composed | QueryFilterCollection([QueryFilter(**empty_json_filter)]).compose()
+                    _exclusion_composed = _exclusion_composed & _filt_composed
+            if _exclusion_composed:
+                _exclusion_composed = (
+                    _exclusion_composed | QueryFilterCollection([QueryFilter(**empty_json_filter)]).compose()
                 )
         else:
-            if tag_filter_list:
-                tag_filter_list.append(empty_json_filter)
-            # We use OR here for our non ocp tables because the tag keys will not live in the
+            if _filter_list:
+                _filter_list.append(empty_json_filter)
+            # We use OR here for our non ocp tables because the  keys will not live in the
             # same json structure.
-            tag_exclusion_composed = self._set_or_filters(tag_filter_list)
-        if tag_exclusion_composed and null_composed:
-            tag_exclusion_composed = tag_exclusion_composed | null_composed
-        return tag_exclusion_composed
+            _exclusion_composed = self._set_or_filters(_filter_list)
+        if _exclusion_composed and null_composed:
+            _exclusion_composed = _exclusion_composed | null_composed
+        return _exclusion_composed
 
-    def _set_tag_filters(self, filters):
-        """Create tag_filters."""
-        tag_column = self._mapper.tag_column
-        tag_filters = self.get_tag_filter_keys()
-        tag_group_by = self.get_tag_group_by_keys()
-        tag_filters.extend(tag_group_by)
-        tag_filters = [tag for tag in tag_filters if "and:" not in tag and "or:" not in tag]
-        for tag in tag_filters:
+    def _build_prefix_filters(self, filters, prefix, db_column):
+        """Create prefix filters.
+
+        filters: FilterCollection
+        prefix: param prefix eg (tag:, aws_category:)
+        db_column: column to build the filter with
+        """
+        if prefix == TAG_PREFIX:
+            _filters = self.get_tag_filter_keys()
+            _group_by = self.get_tag_group_by_keys()
+        elif prefix == AWS_CATEGORY_PREFIX:
+            _filters = self.get_aws_category_keys("filter")
+            _group_by = self.get_aws_category_keys("group_by")
+        else:
+            return filters
+        _filters.extend(_group_by)
+        _filters = [filt for filt in _filters if "and:" not in filt and "or:" not in filt]
+        for filter in _filters:
```

Comment:
Oh dear, so many `filter`, `_filters`, `filt`, `filter_`... BUT `filter` is indeed a python keyword, so we should change this to `for _filter in _filters`.